src/arc.ts

```diff
@@ -195,27 +235,27 @@ export async function initializeArc(provider?: any): Promise<boolean> {
  * @param provider web3Provider
  * @return the expected network nameif not correct
 */
-async function ensureCorrectNetwork(provider: any): Promise<void> {
+async function ensureCorrectNetwork(provider: any, network?: Networks): Promise<void> {
 
   /**
    * It is required that the provider be the correct one for the current platform
    */
-  const expectedNetworkName = targetedNetwork();
+  const expectedNetworkNames = network?[network]: targetNetworks();
 
   // TODO: we should not use the network NAME but the network ID to identify the network...
   const networkName = await getProviderNetworkName(provider);
 
-  if (networkName !== expectedNetworkName) {
-    if (expectedNetworkName === "xdai") {
+  if (!expectedNetworkNames.includes(networkName as Networks)) {
+    if (expectedNetworkNames.includes("xdai")) {
       // TODO: xdai is reporting network 'unknown (100)` , it seems
       if (networkName === "unknown (100)") {
         // we are fine, mayby
         return;
       }
     }
     // eslint-disable-next-line no-console
-    console.error(`connected to the wrong network, should be ${expectedNetworkName} (instead of "${networkName}")`);
-    throw new Error(`Please connect your wallet provider to ${expectedNetworkName}`);
+    console.error(`connected to the wrong network, should be ${expectedNetworkNames} (instead of "${networkName}")`);
+    throw new Error(`Please connect your wallet provider to ${(expectedNetworkNames as Array<any>).join(" or ")}`);
```

Comment:
You are trhowing new error like this here:
```ts
throw new Error('');
```
but previously it was done like this:
```ts
throw Error('');
```
I recommend to use just only first variant everywhere to keep codebase consistent and easy to reead. And yes, I know, that both variants works the same, but code consistency is very important for such huge project

Reply:
This is not related to this PR, we can do a separate code improvements. Please let's focus on changes related to this PR (otherwise we have many many things to fix in Alchemy code 😁)
You already commented so I'll make the change. So I didn't get you, you want to go with or without the `new`?

Reply:
You just use different approaches withing this exact PR